crates/bevy_ecs/src/change_detection.rs

```diff
@@ -5,6 +5,12 @@ use crate::{component::ComponentTicks, system::Resource};
 use bevy_reflect::Reflect;
 use std::ops::{Deref, DerefMut};
 
+/// The minimum number of tick increments between consecutive check tick scans. The maximum is twice this number.
+pub const CHANGE_DETECTION_CHECK_THRESHOLD: u32 = u32::MAX / 2048;
```

Comment:
This needs some explanation of why this value is set to this.

Reply:
Short answer is the choice is arbitrary and the reasoning is a bit circular.

Imagine that the scan threshold is `N` (where `N > 0`) and that we have an app with two stages—the first having `N - 1` systems and the second having `N` systems.

The first stage can increment the world counter at most `N - 1` times, which is less than `N` and won't trigger a scan. The second stage can increment the counter another `N` times before the scan happens.

If changes can age by up to `2 * N - 1` ticks between scans, any ticks older than `UINT::MAX - (2 * N - 1)` are at risk of their wrapping subtraction overflowing.

Should I just copy-paste that as the comment?

Reply:
The threshold `N` doesn't have to be `u32::MAX / 2048`, we could use a nice round number like 1 million.

Also hypothetically, exclusive systems and chain systems can circumvent that because they can increment the world tick more than once. But realistically, nobody will have anything close to `N` systems, and we could always limit the system capacity to be much smaller than `N`.

Reply:
> Should I just copy-paste that as the comment?

Yes please!